pkg/fanal/artifact/local/fs.go

```diff
@@ -98,36 +105,72 @@ func NewArtifact(rootPath string, c cache.ArtifactCache, w Walker, opt artifact.
 	return art, nil
 }
 
-// gitCommitHash returns the latest commit hash if the git repository is clean, otherwise returns an error
-func gitCommitHash(dir string) (string, error) {
+// extractGitInfo extracts git repository information including clean status and metadata
+// Returns clean status (for caching), metadata, and error
+func extractGitInfo(dir string) (bool, artifact.RepoMetadata, error) {
+	var metadata artifact.RepoMetadata
+
 	repo, err := git.PlainOpen(dir)
 	if err != nil {
-		return "", xerrors.Errorf("failed to open git repository: %w", err)
+		return false, metadata, xerrors.Errorf("failed to open git repository: %w", err)
 	}
 
-	// Get the working tree
-	worktree, err := repo.Worktree()
+	// Get HEAD commit
+	head, err := repo.Head()
 	if err != nil {
-		return "", xerrors.Errorf("failed to get worktree: %w", err)
+		return false, metadata, xerrors.Errorf("failed to get HEAD: %w", err)
 	}
 
-	// Get the current status
-	status, err := worktree.Status()
+	commit, err := repo.CommitObject(head.Hash())
 	if err != nil {
-		return "", xerrors.Errorf("failed to get status: %w", err)
+		return false, metadata, xerrors.Errorf("failed to get commit object: %w", err)
 	}
 
-	if !status.IsClean() {
-		return "", xerrors.New("repository is dirty")
+	// Extract basic commit metadata
+	metadata.Commit = head.Hash().String()
+	metadata.CommitMsg = strings.TrimSpace(commit.Message)
+	metadata.Author = commit.Author.String()
+	metadata.Committer = commit.Committer.String()
+
+	// Get branch name
+	if head.Name().IsBranch() {
+		metadata.Branch = head.Name().Short()
 	}
 
-	// Get the HEAD commit hash
-	head, err := repo.Head()
+	// Get tag name if HEAD points to a tag
+	tags, err := repo.Tags()
+	if err == nil {
+		tags.ForEach(func(tag *plumbing.Reference) error {
+			if tag.Hash() == head.Hash() {
+				metadata.Tag = tag.Name().Short()
+				return nil
```

Comment:
I wonder if should add all tags here while we're at it?

Reply:
This field is intended to store information if the current HEAD has a tag, and we would like to keep it as is. However, if the goal is to retrieve all git tags in the repository, we can add a separate field for that purpose.

Reply:
I meant if the current HEAD has multiple tags

<img width="734" height="332" alt="image" src="https://github.com/user-attachments/assets/8744e668-837c-4c86-b09a-3f2315b70d5d" />

Reply:
Ah, yes, we should take care of the case. I missed that. Thanks.

Reply:
technicall the branch can have the same issue

<img width="894" height="178" alt="image" src="https://github.com/user-attachments/assets/a7214ecd-20cd-45a4-bedb-608ba5748a5c" />